builtin/mainmenu/tab_settings_simple.lua

```diff
@@ -16,6 +16,41 @@
 --with this program; if not, write to the Free Software Foundation, Inc.,
 --51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
+local function create_confirm_reset_dlg()
+	return dialog_create("reset_all_settings",
+		function()
+			return
+				"image_button[2,1;8,3;" .. core.formspec_escape(defaulttexturedir ..
+					"blank.png") .. ";;" .. fgettext("Reset all settings?") ..
+					";true;false;]" ..
+				"style[reset_confirm;bgcolor=red]" ..
+				"button[3,4.8;3,0.5;reset_confirm;" .. fgettext("Reset") .. "]" ..
+				"button[6,4.8;3,0.5;reset_cancel;" .. fgettext("Cancel") .. "]"
+		end,
+		function(this, fields)
+			if fields["reset_confirm"] then
+				-- TODO: Maybe only reset the settings that are shown in the
+				-- dialog
```

Comment:
It is best to reset all settings in case the player manually edited .conf, broke something and does not understand what to do.
Or if some game quietly overwrote the settings.